File: app/src/main/java/ska/sudoku/Cell.java
package ska.sudoku;

class Cell {

    private int nr;
    private int value;
    private boolean isPreFilled = false;

    Cell(int nr) {
        this.nr = nr;
        value = 0;
    }

    int getNr() {
        return nr;
    }

    int getValue() {
        return value;
    }

    void setValue(int value) {
        this.value = value;
    }

    boolean isPreFilled() {
        return isPreFilled;
    }

    void setPreFilled(int value) {
        isPreFilled = value != 0;
        setValue(value);
    }

    @Override
    public String toString() {
        return Integer.toString(value);
    }
}

File: app/src/main/java/ska/sudoku/Cell.kt
package ska.sudoku

class Cell internal constructor(internal val nr: Int) {
    enum class State { EMPTY, USER, SOLVED }

    var value: Int = 0
    var state = State.EMPTY

    fun prefill(value: Int) {
        state = if (value == 0) State.EMPTY else State.USER
        this.value = value
    }

    fun markAsSolved() {
        state = State.SOLVED
    }

    override fun toString() = value.toString()
}
